commands.go

```diff
@@ -564,33 +564,45 @@ func (i cJPEG) Init(m *disgomux.Mux) {
 }
 
 func (i cJPEG) Handle(ctx *disgomux.Context) {
-	messages, err := ctx.Session.ChannelMessages(ctx.Message.ChannelID, 2, ctx.Message.ID, "", "")
-	if err != nil {
-		i.issue(err, ctx)
-		return
-	}
-
-	var lastAttachment *discordgo.MessageAttachment
+	var message *discordgo.Message
 
-	messageLen := len(messages)
-	for i := range messages {
-		message := messages[messageLen-1-i]
+	if len(ctx.Arguments) == 0 {
+		messages, err := ctx.Session.ChannelMessages(
+			ctx.Message.ChannelID, 2, ctx.Message.ID, "", "",
+		)
+		if err != nil {
+			i.issue(err, ctx)
+			return
+		}
 
-		if len(message.Attachments) > 0 {
-			for _, attachment := range message.Attachments {
-				lastAttachment = attachment
-				break
-			}
+		messageLen := len(messages)
+		for i := range messages {
+			message = messages[messageLen-1-i]
+		}
+	} else {
+		var err error
+		message, err = ctx.Session.ChannelMessage(
+			ctx.Message.ChannelID, ctx.Arguments[0],
+		)
+		if err != nil {
+			ctx.ChannelSend(
+				fmt.Sprintf(
+					"No message with ID `%s` found in this channel.",
+					ctx.Arguments[0],
+				),
+			)
+			return
 		}
 	}
 
-	if lastAttachment == nil {
-		ctx.ChannelSend("Couldn't find valid image")
+	if len(message.Attachments) == 0 || message.Attachments[0] == nil {
+		ctx.ChannelSend("The message doesn't have any attachments.")
 		return
 	}
 
-	if strings.HasSuffix(lastAttachment.ProxyURL, ".png") || strings.HasSuffix(lastAttachment.ProxyURL, ".jpg") {
-		req, err := http.Get(lastAttachment.ProxyURL)
+	attachment := message.Attachments[0]
+	if strings.HasSuffix(attachment.ProxyURL, ".png") || strings.HasSuffix(attachment.ProxyURL, ".jpg") {
```

Comment:
May want to add support for other valid jpeg file extensions. I'd say .jpeg at least, since it's somewhat common. .jpe, .jfif, and .jif if you are feeling overadventurous.

Reply:
Good point, I'll definitely add that.